go/performance/benchmarks/dataset.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"log"
+	"strings"
+)
+
+// Dataset is a set of test data used for benchmark testing
+type Dataset interface {
+	// String returns a string formatted based on the Dataset's SeedSchema
+	String() string
+
+	// Bytes returns a byte slice formatted based on the Dataset's SeedSchema
+	Bytes() []byte
+
+	// Change returns a Dataset mutated by the given percentage of change
+	Change(pct float32) Dataset
+}
+
+// DSImpl implements the Dataset interface
+type DSImpl struct {
+	// Schema defines the structure of the Dataset
+	Schema *SeedSchema
+
+	// TableName is the name of the test dataset
+	TableName string
+}
+
+// NewDSImpl creates a new DSImpl
+func NewDSImpl(sch *SeedSchema, tableName string) *DSImpl {
```

Comment:
Future extension: I would probably make this write to an `io.Writer` instead, so that it doesn't have to create a string of the entire dataset in memory at once.